### ISSUE TITLE
Adding Metrics to the package-table from the 'DataONE Metrics Service'

### ISSUE THREAD

**Issue by rushirajnenuji** [NCEAS/metacatui#795]
Adding Metrics to the package-table from the 'DataONE Metrics Service'. This is an extension to Issue #784

**Comment by rushirajnenuji** [comment]
Solved by replaced the metrics in the package table to query the `DataONE metrics-service`.

Relevant commits: https://github.com/NCEAS/metacatui/commit/0b3e1def13a5d665f980f7cf8e9e768ee6a1ba5b

**Comment by aebudden** [comment]
I found an example where the downloads don't match between the table and the pills (https://search.dataone.org/view/urn:uuid:5ec25285-9452-4a96-9f8f-0808715722e7). Only one I have found so far but wasn't sure if there was a reason for this.

**Comment by rushirajnenuji** [comment]
Hey @aebudden - thank's for bringing up this issue. Yes, this is an edge-case where the summation of metrics of objects in Package Table won't match the one with the Summary within the DataONE space.

TL;DR: The files that have `non-zero` counts belong to the `obsoleted` packages. Since, according to `COUNTER`, we aggregate metrics based on previous versions on the dataset level, we count the stats of those objects (~which could have been removed from the updated package). As these objects don't belong to the current version (in our case ` resource_map_urn:uuid:7bedc5bc-3e73-4ad4-968d-ed5b6999190b`), MetacatUI does not list those objects in Package table. 

Description: 

From the obsolescence relations available to us from the SOLR endpoint, here is what we have.
(MOST recently updated version at the top). The indented identifiers are the documents of that particular `resourceMap`.

```
 resource_map_urn:uuid:7bedc5bc-3e73-4ad4-968d-ed5b6999190b
	<< Objects >>
	urn:uuid:4e0f5fa8-b958-4d7f-b245-367bff3a5204
	urn:uuid:fa1320c0-7060-4ba7-8e2a-26e829817c77
	urn:uuid:5ec25285-9452-4a96-9f8f-0808715722e7
	urn:uuid:05768bed-109f-42e5-b420-1f2a34ded035

 resource_map_urn:uuid:a0116eb0-9d4f-4f32-8797-a8a89b9ae769:
	<< Objects >>
	urn:uuid:e8d00392-45c0-49ca-a79a-20f9ed047768
	urn:uuid:ae17863e-607f-478f-b544-629f88e8a764
	urn:uuid:4e0f5fa8-b958-4d7f-b245-367bff3a5204
	urn:uuid:5ec25285-9452-4a96-9f8f-0808715722e7

 resource_map_urn:uuid:6d0acf93-2aec-46ca-a451-e7ee0b9ec43b:
	<< Object >>
	urn:uuid:5ec25285-9452-4a96-9f8f-0808715722e7

urn:uuid:659e931c-f98d-44d2-8221-b5591d493cf6: 
	<< Objects >>
	urn:uuid:4e0f5fa8-b958-4d7f-b245-367bff3a5204
	urn:uuid:fa1320c0-7060-4ba7-8e2a-26e829817c77
	urn:uuid:5ec25285-9452-4a96-9f8f-0808715722e7
	urn:uuid:05768bed-109f-42e5-b420-1f2a34ded035
```

The `DataONE Metrics Service` returns the results as:
```
"metrics_package_counts": {
            "urn:uuid:05768bed-109f-42e5-b420-1f2a34ded035": {
                "downloads": 0,
                "views": 0
            },
            "urn:uuid:e8d00392-45c0-49ca-a79a-20f9ed047768": {
                "downloads": 2,
                "views": 0
            },
            "urn:uuid:5ec25285-9452-4a96-9f8f-0808715722e7": {
                "downloads": 0,
                "views": 30
            },
            "urn:uuid:4e0f5fa8-b958-4d7f-b245-367bff3a5204": {
                "downloads": 1,
                "views": 0
            },
            "urn:uuid:ae17863e-607f-478f-b544-629f88e8a764": {
                "downloads": 2,
                "views": 0
            },
            "urn:uuid:fa1320c0-7060-4ba7-8e2a-26e829817c77": {
                "downloads": 0,
                "views": 0
            }
        }
```

Upon further investigation, the object `urn:uuid:e8d00392-45c0-49ca-a79a-20f9ed047768` that has 2 counts was registered to DataONE as `Cons_Atlas_DwC_fields_180929.xlsx` but in the updated / most recent package the object `urn:uuid:4e0f5fa8-b958-4d7f-b245-367bff3a5204` is registered as `036.dwc_obs_2018.csv`. There are two cases possible - that the user might have just renamed the file or might have deleted it and replaced with a different one!  DataONE treats them as two separate objects and hence does not have any relation between `Cons_Atlas_DwC_fields_180929.xlsx` and `036.dwc_obs_2018.csv`.

Here's a pretty similar scenario that I discussed with Chris and Lauren, a couple of days ago [Document](https://hpad.dataone.org/EYBgxmAsBmCGBsBaAptZB2RMAcnsBN9ZF95YAmSckAVnhoGZkg==)

If my write up is confusing, then I'm happy to go over this on the call.

**Comment by laurenwalker** [comment]
The only thing left to do here is make sure that the metrics being loaded into the table are not holding up the entire table from being rendered. The table should be inserted into the page with the metric counts missing, if the metrics request is still pending. Once the MetricModel is synced, the metrics column in the table can be updated with the counts.

Assigning to 2.3.2 milestone

**Comment by rushirajnenuji** [comment]
Solved with commits https://github.com/NCEAS/metacatui/commit/713a04e04d554d570142c7de6c9e61ec1e73964c and https://github.com/NCEAS/metacatui/commit/29bf1c36edae83f0cffd285e3a86c4b6df06f483

- Display metrics if they've already been loaded while loading the Package table, else we continure the rendering and update the metrics later on in the app life-cycle.
